### PR TITLE
pygmt.config: Correctly resetting to default values that contain whitespaces

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Fixes https://github.com/GenericMappingTools/pygmt/issues/2298

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_config.py [hunk 1]
```diff
@@ -72,8 +72,13 @@ def test_config_format_date_map():
 
     Note the space in 'o dd', this acts as a regression test for
     https://github.com/GenericMappingTools/pygmt/issues/247.
+
+    Setting FORMAT_DATE_MAP="yyyy mm dd" as a regression test for
+    https://github.com/GenericMappingTools/pygmt/issues/2298.
     """
     fig = Figure()
+    # Set the FORMAT_DATE_MAP to "yyyy mm dd" that contains whitespaces.
```
Comment (seisman) [inline pygmt/tests/test_config.py:80]:
```suggestion
    # Set FORMAT_DATE_MAP to "yyyy mm dd" which contains whitespaces.
```

File: pygmt/tests/test_config.py [hunk 2]
```diff
@@ -72,8 +72,13 @@ def test_config_format_date_map():
 
     Note the space in 'o dd', this acts as a regression test for
     https://github.com/GenericMappingTools/pygmt/issues/247.
+
+    Setting FORMAT_DATE_MAP="yyyy mm dd" as a regression test for
+    https://github.com/GenericMappingTools/pygmt/issues/2298.
     """
     fig = Figure()
+    # Set FORMAT_DATE_MAP to "yyyy mm dd" which contains whitespaces.
+    config(FORMAT_DATE_MAP="yyyy mm dd")
```
Comment (willschlitzer) [inline pygmt/tests/test_config.py:81]: Is a test being added to use this whitespace-containing `FORMAT_DATE_MAP` or is this sufficient to be used on other tests to make sure that whitespace can work?

**Review by weiji14 — APPROVED** [review]
Thanks @seisman, good catch on finding the whitespace issue during the `pygmt.config` reset, should have caught this earlier in #1487.

Oh, and Happy Chinese New Year :rabbit2::firecracker:

**Review by michaelgrund — APPROVED** [review]

File: pygmt/tests/test_config.py (hunk 2 above)
Comment (seisman) [reply pygmt/tests/test_config.py:81]: > is this sufficient to be used on other tests to make sure that whitespace can work?

Yes, please see https://github.com/GenericMappingTools/pygmt/issues/2298#issuecomment-1396555512

**Pull request merged**